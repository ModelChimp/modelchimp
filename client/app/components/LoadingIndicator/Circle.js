import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'styled-components';

const circleFadeDelay = keyframes`
  0%,
  39%,
  100% {
    opacity: 0;
  }

  40% {
    opacity: 1;
  }
`;

const circleFadeDelayRule = css`
  ${circleFadeDelay} 1.2s infinite ease-in-out both;
`;

const Circle = props => {
  const CirclePrimitive = styled.div`
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    ${props.rotate &&
      `
      -webkit-transform: rotate(${props.rotate}deg);
      -ms-transform: rotate(${props.rotate}deg);
      transform: rotate(${props.rotate}deg);
    `} &:before {
      content: '';
      display: block;
      margin: 0 auto;
      width: 15%;
      height: 15%;
      background-color: #999;
      border-radius: 100%;
      animation: ${circleFadeDelayRule};
      ${props.delay &&
        `
        -webkit-animation-delay: ${props.delay}s;
        animation-delay: ${props.delay}s;
      `};
    }
  `;
  return <CirclePrimitive />;
};

Circle.propTypes = {
  delay: PropTypes.number,
  rotate: PropTypes.number,
};

export default Circle;
