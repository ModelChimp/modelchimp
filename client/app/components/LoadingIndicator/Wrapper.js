import styled from 'styled-components';

const Wrapper = styled.div`
  margin: 2em auto;
  width: 40px;
  height: 40px;
  position: relative;
`;

export default Wrapper;
