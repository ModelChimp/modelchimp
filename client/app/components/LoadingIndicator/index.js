import React from 'react';

import Circle from './Circle';
import Wrapper from './Wrapper';

const LoadingIndicator = () => (
  <Wrapper>
    <Circle />
    <Circle rotate={30} delay={-1.1} />
    <Circle rotate={60} delay={-1} />
    <Circle rotate={90} delay={-0.9} />
    <Circle rotate={120} delay={-0.8} />
    <Circle rotate={150} delay={-0.7} />
    <Circle rotate={180} delay={-0.6} />
    <Circle rotate={210} delay={-0.5} />
    <Circle rotate={240} delay={-0.4} />
    <Circle rotate={270} delay={-0.3} />
    <Circle rotate={300} delay={-0.2} />
    <Circle rotate={330} delay={-0.1} />
  </Wrapper>
);

export default LoadingIndicator;
