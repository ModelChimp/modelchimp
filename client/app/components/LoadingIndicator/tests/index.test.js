import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import LoadingIndicator from '../index';

describe('<LoadingIndicator />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<LoadingIndicator />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
