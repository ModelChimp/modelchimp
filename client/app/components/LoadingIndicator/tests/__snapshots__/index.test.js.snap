// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should match the snapshot 1`] = `
.c0 {
  margin: 2em auto;
  width: 40px;
  height: 40px;
  position: relative;
}

.c1 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
}

.c1:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
}

.c2 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
}

.c2:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -1.1s;
  -webkit-animation-delay: -1.1s;
  animation-delay: -1.1s;
}

.c3 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
}

.c3:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -1s;
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

.c4 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.9s;
  -webkit-animation-delay: -0.9s;
  animation-delay: -0.9s;
}

.c5 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
}

.c5:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.8s;
  -webkit-animation-delay: -0.8s;
  animation-delay: -0.8s;
}

.c6 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
}

.c6:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.7s;
  -webkit-animation-delay: -0.7s;
  animation-delay: -0.7s;
}

.c7 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c7:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.6s;
  -webkit-animation-delay: -0.6s;
  animation-delay: -0.6s;
}

.c8 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
}

.c8:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.5s;
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c9 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
}

.c9:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.4s;
  -webkit-animation-delay: -0.4s;
  animation-delay: -0.4s;
}

.c10 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.c10:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.3s;
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c11 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
}

.c11:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.2s;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c12 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
}

.c12:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: evVlQQ 1.2s infinite ease-in-out both;
  animation: evVlQQ 1.2s infinite ease-in-out both;
  -webkit-animation-delay: -0.1s;
  -webkit-animation-delay: -0.1s;
  animation-delay: -0.1s;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
</div>
`;
